software/winrm/src/main/java/org/apache/brooklyn/util/core/internal/winrm/winrm4j/Winrm4jTool.java

```diff
@@ -75,7 +78,11 @@ public Winrm4jTool(Map<String,?> config) {
     public Winrm4jTool(ConfigBag config) {
         this.bag = checkNotNull(config, "config bag");
         host = getRequiredConfig(config, PROP_HOST);
-        port = getRequiredConfig(config, PROP_PORT);
+        port = config.get(PROP_PORT.getConfigKey()) != null
```

Comment:
The logic I follow here is:
When initializing the `WinRmTool` this could be set explicitly from the WinRmTool properties
however if not the default value varies to be either 5985 or 5986.
It cannot be set from https://github.com/bostko/brooklyn-server/blob/upgrade_winrm4j/software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java#L324
The most proper way is to set this from the entity as we already have this sensor set with this logic in the VanillaSoftwareProcess, however there is no link between the entity and the machine location so I do this check `config.get(USE_SECURE_WINRM) ` once again.